log/hooks.go

```diff
@@ -0,0 +1,72 @@
+package log
+
+import (
+	"fmt"
+	bs "github.com/bugsnag/bugsnag-go"
+	"github.com/docker/machine/log/bugsnag"
+	"github.com/docker/machine/version"
+	"net"
+	"os"
+	"runtime"
+	"strings"
+)
+
+// Empty logger for bugsnag
+type void struct {
+}
+
+func (d *void) Printf(fmtString string, args ...interface{}) {
+}
+
+// SetBugsnag configures the bugsnag hook for logrus
+func SetBugsnag(apiKey string) error {
+	stage := fmt.Sprintf("%s (%s)", runtime.GOOS, runtime.GOARCH)
+
+	bs.Configure(bs.Configuration{
+		APIKey: apiKey,
+		// XXX we need to abuse bugsnag metrics to get the OS/ARCH information as a usable filter
+		// Can do that with either "stage" or "hostname"
+		ReleaseStage: stage,
+		//		Hostname: stage,
+		ProjectPackages: []string{"github.com/docker/machine/[^v]*"},
+		AppVersion:      version.Version + " (" + version.GitCommit + ")",
+		Logger:          new(void),
+		Synchronous:     true,
```

Comment:
I don't know much about the motivation for this, but maybe it should be async?  I certainly wouldn't like to block the main machine thread any time we log.